src/comments/comment.service.ts

```diff
@@ -0,0 +1,247 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Answer } from '../answer/answer.entity';
+import { Question } from '../questions/questions.legacy.entity';
+import { User } from '../users/users.legacy.entity';
+import { CommentDto } from './DTO/comment.dto';
+import { GetCommentDetailDto } from './DTO/getCommentDetail.dto';
+import { Comment, UserAttitudeOnComments } from './comment.entity';
+import {
+  CommentNotFoundByUserError,
+  CommentNotFoundError,
+  CommentableIdNotFoundError,
+  InvalidAgreeTypeError,
+} from './comment.error';
+@Injectable()
+export class CommentsService {
+  constructor(
+    @InjectRepository(User)
+    private usersRepository: Repository<User>,
+    @InjectRepository(UserAttitudeOnComments)
+    private userAttitudeOnCommentsRepository: Repository<UserAttitudeOnComments>,
+    @InjectRepository(Comment)
+    private commentsRepository: Repository<Comment>,
+    @InjectRepository(Answer)
+    private answersRepository: Repository<Answer>,
+    @InjectRepository(Question)
+    private questionsRepository: Repository<Question>,
+  ) {}
+
+  async createComment(
+    userId: number,
+    content: string,
+    commentableType: 'answer' | 'comment' | 'question',
+    commentableId: number,
+  ): Promise<number> {
+    let commentableRepository;
+    switch (commentableType) {
+      case 'answer':
+        commentableRepository = this.answersRepository;
+        break;
+      case 'comment':
+        commentableRepository = this.commentsRepository;
+        break;
+      case 'question':
+        commentableRepository = this.questionsRepository;
+        break;
+    }
+    const commentable = await commentableRepository.findOne({
+      where: { id: commentableId },
+    });
+
+    if (!commentable) {
+      throw new CommentableIdNotFoundError(commentableId);
+    }
+
+    const comment = this.commentsRepository.create({
+      userId,
+      content,
+      commentableType,
+      commentableId,
+    });
+    const userAttitudeOnComment = this.userAttitudeOnCommentsRepository.create({
+      agreeType: '3',
```

Comment:
too difficult to understand. Still consider using enum

Reply:
emmmm，the problem is， the enum type cant send the messsage

Reply:
I will try